my-dsl.md

```diff
@@ -1,38 +1,30 @@
 # Language
-_What is the name of the language? Link the name to its webpage 
-(if appropriate)._
+[Protocol Buffers](https://developers.google.com/protocol-buffers/)
 
 
 # Domain
-_Describe the language's domain in five words._
+Sending and processing language-neutral data.
 
 
 # Computational model
-_What is the underlying computational model of this language? To answer this 
-question, provide a high-level description (no more than 100 words) of the 
-computation that occurs when someone executes a program in this language._
+The file that defines the messages is compiled, and then instances of the messages used by other applications are encoded to binary based on the object file.
 
 
 # DSL-ness
-_Fowler writes about a spectrum of languages, from general-purpose languages to 
-"purely" domain-specific. Where does the DSL you chose fall on this spectrum, 
-and why?_ 
+Protocol Buffers falls much closer to the domain-specific side of the spectrum. Fowler often refers to whether a language supports conditional logic and looping to determine whether it's general-purpose, and as far as I know, Protocol Buffers supports neither feature.
 
 
 # Internal or external?
-_Is the language implemented as an internal or external DSL? 
-Justify your answer._
+Protocol Buffers is an external DSL as it does not employ the syntax of the language in which it's written.
 
 
 # Host language
-_What language(s) was (were) used to implement the DSL?_
+There's an implementation of Protocol Buffers for each language the DSL supports: Java, C++, Python, JavaNano, Ruby (and I think others).
 
 
 # Benefits
-_Identify one potential benefit of the DSL: how is a programmer's life or a 
-company's bottom line made easier by the existence of this language?_
+Protocol Buffers allows users to transfer data across applications--potentially, even if the applications are written in separate languages. The DSL also generates classes for the defined messages to more easily set and access the message fields.
 
 
 # Drawbacks
-_Identify one potential drawback of the DSL: what does a programmer or company 
-lose by using this DSL instead of a general-purpose language?_
+With respect to XML, Protocol Buffers is not designed to model structure and is not as human-readable.
```

Comment:
It seems like another drawback is the fact that you need the class definition to completely understand a protocol buffer while when you receive XML, you can interpret everything from just the information sent.